src/librustdoc/passes/html_tags.rs

```diff
@@ -94,6 +94,14 @@ fn extract_path_backwards(text: &str, end_pos: usize) -> Option<usize> {
     if current_pos == end_pos { None } else { Some(current_pos) }
 }
 
+fn is_valid_for_html_tag_name(c: char, is_empty: bool) -> bool {
+    // https://spec.commonmark.org/0.30/#raw-html
```

Comment:
This norm is a bit weird since it allows stuff like `a-------------------`. Oh well, not our problem. :)